src/docker/README.md

```diff
@@ -1,26 +1,57 @@
-# Quick test environment setup using Docker
-
 In order to test the plugin on different versions of `GitLab` and `Jenkins` you may want to use `Docker` containers.
 
-A example docker-compose file is available at `gitlab-plugin/src/docker` which allows to set up instances of the latest `GitLab` and `Jenkins` versions.
+# Quick test environment setup using Docker for Linux/amd64
+
+An example docker-compose file is available at `gitlab-plugin/src/docker` which the user can use to set up instances of the latest `GitLab` version and latest `Jenkins` LTS version for linux/amd64.
 
 If they don't already exist, create the following directories and make sure the user that Docker is running as owns them:
 * /srv/docker/gitlab/postgresql
 * /srv/docker/gitlab/gitlab
 * /srv/docker/gitlab/redis
 * /srv/docker/jenkins
+To start the containers for Linux, run `docker-compose up -d` from the `docker` folder. If you have problems accessing the services in the containers, run `docker-compose up` by itself to see output from the services as they start, and the latter command is the verbose version of the former.
+
+## Quick test environment setup using Docker for MacOS/arm64
+
+You need to modify the example docker-compose file available at `gitlab-plugin/src/docker` to set up instances of the latest `GitLab` and `Jenkins` versions for MacOS/arm64. 
 
-To start the containers, run `docker-compose up -d` from the `docker` folder. If you have problems accessing the services in the containers, run `docker-compose up` by itself to see output from the services as they start.
+Due to Apple's System Integrity Protection (SIP), the suggested paths cannot be simply created and accessed, so you may need to use the home directory (~) as a root for the new directories to be created.
+
+In the `docker-compose.yml` file :
+    1. Change the ports to 
+       - '55580:80'
+       - '55522:22'
+       - '55443:443'
+      as the browser may block the ports in original docker-compose file.
+    2. Change the gitlab volumes to 
+        `/Users/yourusername/srv/docker/gitlab/config:/etc/gitlab`
+        `/Users/yourusername/srv/docker/gitlab/logs:/var/log/gitlab`
+        `/Users/yourusername/srv/docker/gitlab/data:/var/opt/gitlab`
+    3. Change the jenkins volumes to 
+        `/Users/yourusername/srv/docker/jenkins:/var/jenkins_home`
+    4. In your Docker-Desktop go to `Settings > General > Choose file sharing implementation for your containers` and switch to osxfs (Legacy). As `osxfs (Legacy)` utilizes more resources of the system, make sure the assigned resources are sufficient by going to `Settings > Resources` and make suitable adjustments where necessary, otherwise Docker Desktop may go on start mode forever on restarting.
```

Comment:
```suggestion
4. In your Docker-Desktop go to `Settings > General > Choose file sharing implementation for your containers` and switch to osxfs (Legacy). As `osxfs (Legacy)` utilizes more resources of the system, make sure the assigned resources are sufficient by going to `Settings > Resources` and make suitable adjustments where necessary, otherwise Docker Desktop may go on start mode forever on restarting.
```